tensorboard/webapp/tbdev_upload/tbdev_upload_dialog_component.ts

```diff
@@ -12,20 +12,62 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 See the License for the specific language governing permissions and
 limitations under the License.
 ==============================================================================*/
-import {Component} from '@angular/core';
+import {Component, OnInit, OnDestroy} from '@angular/core';
 import {MatDialogRef} from '@angular/material/dialog';
+import {Store, select} from '@ngrx/store';
+import {State, getEnvironment} from '../core/store';
+import {Subject} from 'rxjs';
+import {takeUntil} from 'rxjs/operators';
+
+/** @typehack */ import * as _typeHackRxjs from 'rxjs';
 
 @Component({
   selector: 'tbdev-upload-dialog',
   templateUrl: './tbdev_upload_dialog_component.ng.html',
   styleUrls: ['./tbdev_upload_dialog_component.css'],
 })
-export class TbdevUploadDialogComponent {
-  readonly commandText: string = 'tensorboard dev upload --logdir {logdir}';
+export class TbdevUploadDialogComponent implements OnInit, OnDestroy {
+  commandText: string = this.getCommandText('');
+  readonly environment$ = this.store.pipe(select(getEnvironment));
+
+  private ngUnsubscribe = new Subject();
+
+  constructor(
+    private dialogRef: MatDialogRef<TbdevUploadDialogComponent>,
+    private store: Store<State>
+  ) {}
+
+  ngOnInit() {
+    this.environment$
+      .pipe(takeUntil(this.ngUnsubscribe))
+      .subscribe((environment) => {
+        if (environment && environment.data_location) {
+          this.commandText = this.getCommandText(environment.data_location);
+        }
+      });
```

Comment:
A component should not have access to the store. Please create a container and pass down the environment as input using the async pipe.

Reply:
Done.  I've tried to mimic what is being done in header/plugin_selector.container.ts.  Note that I originally mimicked what is being done in settings/dialog_component.ts so I don't know if we should update that component as well.